### DISCUSSION TITLE
Migrating existing applications to Quarkus with Migration Toolkit for Applications

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#38129] [Quarkus Blog/Website]
# Migrating existing applications to Quarkus with Migration Toolkit for Applications

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/migrating-existing-apps-to-quarkus-with-mta/

**Comment by ymajoros** [comment]
> JBoss Web Server (the Red Hat Supported build of Tomcat)
JBoss isn't a build of Tomcat, at all. It's an application server implementing Java (now Jakarta) EE implementation, while Tomcat is a Servlet container.

JBoss is more a "Red Had build of Wildfly" (with support, which means if you are patient enough and open, create a lot of support tickets and are generally annoying, they will maybe backport bugfixes of Wildfly long after they are done).

**Reply by wteschke** [reply]
You are right that JBoss EAP is based on wildfly, the JBoss web server however seems to be a different thing, just follow the link above to read more about it.

**Reply by gsmet** [reply]
> with support, which means if you are patient enough and open, create a lot of support tickets and are generally annoying, they will maybe backport bugfixes of Wildfly long after they are done

(Disclaimer: I'm from Red Hat)
I don't think that's very fair. We are extremely cautious when backporting things to supported versions as people expect stability over all. And even the smallest patch can have some weird and unexpected consequences.
So we usually want to make sure there is no workaround and the problem is important enough and also that it had some bake time in the latest versions before backporting.

In the case of Quarkus, you can even see that in the community LTS: patches arrive later and we don't release as often as the latest stream. Because we don't want to break people using the LTS.
And even doing that, we got hit a few times by unexpected side effects.
